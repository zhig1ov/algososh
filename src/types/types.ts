import { ElementStates } from "./element-states"

export interface listItemProps {
  adding?: boolean;
  deleting?: boolean;
  noArrow?: boolean;
  tail?: string;
  head?: string;
  char?: string;
  state: ElementStates;
  extraCircle?: {
    char: string;
  }
}

export interface columnObject {
  num: number;
  state: ElementStates;
}

export type radioState = "selection" | "bubble"