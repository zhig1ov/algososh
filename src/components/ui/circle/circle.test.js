import React from "react"
import renderer from "react-test-renderer"
import { Circle } from "./circle"
import { ElementStates } from "../../../types/element-states"

describe('Работы компонента Circle', () => {
  it('Circle без буквы работает без ошибок', () => {
    const tree = renderer
			.create(<Circle  />)
			.toJSON()
		expect(tree).toMatchSnapshot()
  })

  it('Circle с буквой работает без ошибок', () => {
		const tree = renderer
			.create(<Circle letter='ABCD'/>)
			.toJSON()
		expect(tree).toMatchSnapshot()
  })

  it('Circle с head работает без ошибок', () => {
		const tree = renderer
			.create(<Circle head='string'/>)
			.toJSON()
		expect(tree).toMatchSnapshot()
  })

	it('Circle с react элементом в head работает без ошибок', () => {
		const tree = renderer
			.create(<Circle head={<Circle />}/>)
			.toJSON()
		expect(tree).toMatchSnapshot()
  })

	it('Circle с tail работает без ошибок', () => {
		const tree = renderer
			.create(<Circle tail='string'/>)
			.toJSON()
		expect(tree).toMatchSnapshot()
  })

	it('Circle с react элементом в tail работает без ошибок', () => {
		const tree = renderer
			.create(<Circle tail={<Circle />}/>)
			.toJSON()
		expect(tree).toMatchSnapshot()
  })

	it('Circle с index работает без ошибок', () => {
		const tree = renderer
			.create(<Circle index={5}/>)
			.toJSON()
		expect(tree).toMatchSnapshot()
  })

	it('Circle с пропом isSmall работает без ошибок', () => {
		const tree = renderer
			.create(<Circle isSmall={true}/>)
			.toJSON()
		expect(tree).toMatchSnapshot()
  })

	it('Circle в состоянии default работает без ошибок', () => {
		const tree = renderer
			.create(<Circle state={ElementStates.Default}/>)
			.toJSON()
		expect(tree).toMatchSnapshot()
  })

	it('Circle в состоянии changing работает без ошибок', () => {
		const tree = renderer
			.create(<Circle state={ElementStates.Changing}/>)
			.toJSON()
		expect(tree).toMatchSnapshot()
  })

	it('Circle в состоянии modified работает без ошибок', () => {
		const tree = renderer
			.create(<Circle state={ElementStates.Modified}/>)
			.toJSON()
		expect(tree).toMatchSnapshot()
  })
})