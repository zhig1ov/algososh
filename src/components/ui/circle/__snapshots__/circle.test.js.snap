// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Работы компонента Circle Circle без буквы работает без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Работы компонента Circle Circle в состоянии changing работает без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Работы компонента Circle Circle в состоянии default работает без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Работы компонента Circle Circle в состоянии modified работает без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Работы компонента Circle Circle с head работает без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    string
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Работы компонента Circle Circle с index работает без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    5
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Работы компонента Circle Circle с react элементом в head работает без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Работы компонента Circle Circle с react элементом в tail работает без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Работы компонента Circle Circle с tail работает без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    string
  </div>
</div>
`;

exports[`Работы компонента Circle Circle с буквой работает без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      ABCD
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Работы компонента Circle Circle с пропом isSmall работает без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
